### PR TITLE
LTS - Improving test coverage

### PR DESCRIPTION
**Pull request by srowhani**
#patch#

# changelog
- Edge cases in terms of usage
- Improve test coverage

### CODE REVIEW

File: tests/unit/components/nav-modal-test.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      // renders the component on the page
+      this.render()
+      expect(component).to.be.ok
+      expect(this.$()).to.have.length(1)
+    })
+    it('handles actions', function () {
+      let component = this.subject()
+      this.render()
+      Object.keys(component._actions).forEach(e => {
+        component.send(e, {})
```
Comment (rox163) [inline tests/unit/components/nav-modal-test.js]: doesnt seem to be a check/assertion here

Comment (rox163) [inline tests/unit/components/nav-modal-test.js]: if you want to check for function calls maybe you could use sinon here to assert that the action was indeed handled?

File: bower.json [hunk 2]
```diff
@@ -17,6 +16,7 @@
     "ember": "2.6.0"
   },
   "devDependencies": {
-    "blanket": "5e94fc30f2e694bb5c3718ddcbf60d467f4b4d26"
+    "blanket": "5e94fc30f2e694bb5c3718ddcbf60d467f4b4d26",
+    "lodash": "^4.15.0"
```
Comment (rox163) [inline bower.json]: if the lodash shim didnt work then move this back into dependencies above.

**Comment by rox163** [comment]
also change to #patch#

File: addon/templates/components/nav-modal.hbs [hunk 3]
```diff
@@ -2,46 +2,15 @@
...
-            </div>
+        {{#each column as |_section|}}
+          {{#if _section.isVisible}}
+            {{nav-section section=_section showMore=(action 'showMore')}}
           {{/if}}
         {{/each}}
       </div>
     {{/each}}
   {{else}}
+    {{log content}}
```
Comment (rox163) [inline addon/templates/components/nav-modal.hbs]: do you need this log in the final version?

**Comment by rox163** [comment]
looks like you need a merge with the latest master

File: addon/components/nav-modal.js [hunk 4]
```diff
@@ -1,36 +1,48 @@
...
 
   classNames: ['nav-modal'],
   layout,
   tabindex: 0,
   attributeBindings: ['tabindex'],
   activeCategory: null,
-  _categoryChanged: Ember.observer('nav._activeCategory', function () {
-    this.set('activeCategory', this.get('nav._activeCategory'))
+  _categoryChanged: observer('frostNavigation._activeCategory', function () {
+    this.set('activeCategory', this.get('frostNavigation._activeCategory'))
```
Comment (sglanzer) [inline addon/components/nav-modal.js]: Is this consumed in a downstream template?  If so, you may want to wrap it in a `scheduleOnce('sync',`

File: addon/components/nav-route.js [hunk 5]
```diff
@@ -2,28 +2,38 @@ import Ember from 'ember'
...
-      this.get('nav').dismiss()
+      navigation.dismiss()
       return
     }
-    this.get('nav').transitionTo(this.get('route'))
+    navigation.transitionTo(this.get('route'))
+  },
+  getDefaultProps () {
+    return {
+      name: null,
```
Comment (sglanzer) [inline addon/components/nav-route.js]: Shouldn't need the nulls

File: tests/integration/components/nav-category-test.js [hunk 6]
```diff
@@ -17,5 +20,24 @@ describeComponent(
...
       expect(this.$()).to.have.length(1)
     })
+    describe('clicks', function () {
+      const helper = function (name, active) {
+        this.set('_name', name)
+        this.set('_nav', Ember.Object.create({
+          _activeCategory: active
+        }))
+        this.render(hbs`{{nav-category name=_name frostNavigation=_nav}}`)
+        this.$('.nav-category').click()
```
Comment (sglanzer) [inline tests/integration/components/nav-category-test.js]: Can you start adding `ember-hook` to this project as well?

File: tests/integration/components/nav-action-test.js [hunk 7]
```diff
@@ -17,5 +17,18 @@ describeComponent(
       this.render(hbs`{{nav-action}}`)
       expect(this.$()).to.have.length(1)
     })
+    it('calls perform action', function () {
+      let self = this
+      let nav = Ember.Object.create({
+        performAction () {
+          self.set('_item', null)
```
Comment (sglanzer) [inline tests/integration/components/nav-action-test.js]: You can use `sinon` to spy on function calls instead of doing this kind of hack

**Comment by sglanzer** [comment]
Awesome initiative to add more tests into this, I think you just need to address @rox163's lodash comment

I added a few suggestions on how to improve the tests, but I'm fine to approve/merge without those

**Comment by rox163** [comment]
did you apply the hook to the nav addon itself? I see it in the component tests but not in the addon itself.  Can a user set the hook on the parent frost-nav component and will it be sent down to the categories etc automatically? Sort example - https://github.com/ciena-frost/ember-frost-sort/pull/23/files

**Comment by srowhani** [comment]
@rox163 Cool. I'll draw up another PR to avoid bloating this one too much.
